app/controllers/api/v1/screenshots_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    # Controller methods to call a service of AWS S3
+    class ScreenshotsController < ApplicationController
+      before_action :product_version, :locale
+
+      def index
+        @screenshots = Screenshot.where(product_version_id: @version, locale_id: @locale)
+        render json: @screenshots, include: :images
+      end
+
+      def new
+        @screenshot = Screenshot.new
+      end
+
+      def create
+        @screenshot = Screenshot.new(screenshot_params)
+        @screenshot.save!
```

Comment:
```ruby

# something like
def create
        @screenshot = Screenshot.new(screensoht_params)
        @screenshot.save!
# specific exception handle
rescue Exceptoin => e
       ....
end
```

Reply:
@sourabh-upadhyay Thanks for the review. Below is the example of rescue exception in rails controller I found. Is it the correct way or is there any other approach to be followed. Please do let me know. 

def create
        @screenshot = Screenshot.new(screensoht_params)
        @screenshot.save!
    rescue ActiveRecord::RecordInvalid
        raise Errors::Invalid.new(screenshot.errors.to_h)
    end
end

Reply:
You do not need to `raise` the exception again, just send the error message or something, that will render to the client. It depends upon what error we want to render from the endpoint.

```ruby
# just a suggestion.
def create
      @screenshot = Screenshot.new(screensoht_params)
      @screenshot.save!
rescue ActiveRecord::RecordInvalid
      render json: { error: "custom error message" }, status: 4XX
end
```

Reply:
@sourabh-upadhyay  Done. Please do review. Reference used : [https://api.rubyonrails.org/classes/ActiveRecord/RecordInvalid.html](url).